File: src/regex_engine/parse/astnode/ESCNode.java
package regex_engine.parse.astnode;

public class ESCNode implements ASTNode{
    // ^ $  \\w  \\s \\d

    private int function;
    public ESCNode(int function){
        this.function = function;
    }

    @Override
    public String toString() {
        if (function == 1)
            return "ESC: ^";
        else if (function == 2)
            return "ESC: $";
        else if (function == 3)
            return "ESC: \\w";
        else if (function == 4)
            return "ESC: \\s";
        else if (function == 5)
            return "ESC: \\d";
        else if (function == 6)
            return "ESC: .";
        else
            return super.toString();
    }

    public int getFunction(){
        return this.function;
    }

    public void setFunction(int function){
        this.function = function;
    }
}

File: src/Regex-engine-Kotlin/src/regex_engine/parse/astnode/ESCNode.kt
package regex_engine.parse.astnode

class ESCNode(
        // ^ $  \\w  \\s \\d

        var function: Int) : ASTNode {

    override fun toString(): String {
        return if (function == 1)
            "ESC: ^"
        else if (function == 2)
            "ESC: $"
        else if (function == 3)
            "ESC: \\w"
        else if (function == 4)
            "ESC: \\s"
        else if (function == 5)
            "ESC: \\d"
        else if (function == 6)
            "ESC: ."
        else
            super.toString()
    }
}
